legacy/signals/signals.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package signals
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+// exitSignals holds a mapping of os.Signal.String() to syscall for all terminating signals.
+// When the Watch() function encounters a signal within this map, it explicitly forces the
+// program to exit with the corrosponding syscall exit code.
+var exitSignals = map[string]syscall.Signal{
+	"hangup":              syscall.SIGHUP,
```

Comment:
Can we have this reversed (use syscall.SIGHUP as the key and "hangup" as the value), or even better, just use an empty interface as the value? It's something of an antipattern to use the String representation of the thing to compare when we have the actual thing itself. It's like comparing 2 integers by their string representation instead of using the int type natively.

Reply:
I think we are confusing [os.Signal](https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/os/exec.go;l=69-72) with [syscall.Signal](https://pkg.go.dev/syscall#Signal). When we observe an incoming OS signal, it's of type: os.Signal (an interface). The way i can identify it is by the unique sig.String() it produces. I map these sig.String() values to syscall.Signals. This way I can gain access to the int exit code.